### PR TITLE
Add column/data zipping functionality

### PR DESCRIPTION
**Pull request by ollietulloch**
This PR adds column zipping functionality to the mapper, where 2 (or more) columns can have their string values split by a given `split_char` and then be zipped together in a prescribed order.

E.g. given this csv data

```
"title","value"
"species,colour,legs","dog,brown,4"
```

and this data:

```yaml
    - column: title
      mappings:
      - field: zipped_field
        zip_order: 1
        split_char: ","
    - column: value
      mappings:
      - field: zipped_field
        zip_order: 2
```

This mapped_line would be produced:

```
{ "zipped_field"=>[["species", "dog"], ["colour", "brown"], ["legs", "4"]],
   :rawtext=>{"title"=>"species,colour,legs", "value"=>"dog,brown,4"}}
```

### CODE REVIEW

**Review by bshand — CHANGES_REQUESTED** [review]
This all looks good to me. I've identified one minor typo.

I don't know whether you want to add any sort of validations (or error handling in the mapping code), to deal with incomplete `zip_order` entries or the absence of a `split_char`.

File: docs/zipped-field-mapping.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+This would result in:
+
+```
+{ "zipped_field"=>[["species", "dog"], ["colour", "brown"], ["legs", "4"]],
+   :rawtext=>{"title"=>"species,colour,legs", "value"=>"dog,brown,4"}}
+```
+
+Reversing the `zip_order` in the mapping would result in:
+```
+{ "zipped_field"=>[["dog", "species""], ["brown", "colour"], ["4", "legs"]],
```
[the remark this answers is not in the document]
Comment (ollietulloch) [reply docs/zipped-field-mapping.md]: Good spot - resolved in 6828784

**Comment by ollietulloch** [comment]
> > This all looks good to me. I've identified one minor typo.
> > I don't know whether you want to add any sort of validations (or error handling in the mapping code), to deal with incomplete `zip_order` entries or the absence of a `split_char`.
> 
> Yep, I thought about this a bit. I think I may just update the code to check for `:zipped_values` and `:split_char` before attempting the splitting and zipping e.g.
> 
> `elsif field_data[:zipped_values].present? && field_data[:split_char].present?`

I've added 58a7b92 which ensures there is enough field config present before attempting to zip.

**Review by bshand — APPROVED** [review]
This all looks good.

**Pull request merged**